### PR TITLE
Helper function mg_construct_msg for generating well-formed HTTP respons...

### PR DESCRIPTION
**Pull request by cdbishop**
I've added a response helper function as I didn't like having to manually format the response (with the \r\n details and that)

Included an example showing how to use it/proof that it works

### CODE REVIEW

File: examples/response/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# 
+# Copyright (c) 2013 No Face Press, LLC
```
Comment (bel2125) [inline examples/response/Makefile]: Please don't copy/paste license statements from other examples.
Use the current year and either your name or "the Civetweb developers".

File: examples/response/Makefile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# License http://opensource.org/licenses/mit-license.php MIT License
+#
+
+#This makefile is used to test the other Makefiles
```
Comment (bel2125) [inline examples/response/Makefile]: Is it?
--> Please don't copy/paste, but write your own (brief) description

**Comment by bel2125** [comment]
I need to check API extensions in civetweb.h very carefully.

The functions do not add more functionality but convenience. I was thinking about restructuring the API - to split it in a low level core API in one header and convenience functions in another header - after the next release.

File: examples/response/Makefile (hunk 1 above)
Comment (cdbishop) [inline examples/response/Makefile]: Apologies. That was me being lazy. Corrected as suggested.

**Comment by cdbishop** [comment]
Understood, I agree that this change doesn't really fit with the other API calls.

I'm happy to close this if there are already plans for providing functionality such as this. Assuming you agree that this would be a beneficial addition.

**Comment by cdbishop** [comment]
Having thought about it a bit more, I'm not sure the benefit this function provides justifies the size of the change... It did end up being larger than I initially anticipated (there may be a better way to implement this,  I've not done much C)

My reasoning for attempting this was I felt that it would be beneficial for users to not have to worry about the HTTP message formatting when writing a response, but OTOH its not hard/complicated to create a correct response with mg_printf, so maybe I just got too carried away with this change :)

What do you think? Is there a benefit to providing a function that hides the HTTP semantics of generating responses?

**Comment by bel2125** [comment]
I think an API that is even one level higher might be more useful:
Instead of `mg_construct_response_msg(conn, http_version="1.1", status_code=200, status_msg="OK", headers, data, size)` probably something like `mg_send_response_ok(conn, data, size)`, where the http_version is determined automatically from the request, the status code 200 is known for _ok, the status msg can be determined from the status code (`mg_get_response_code_text`), and additional headers might be set by something like `mg_set_response_header(conn, field_name, value)`. Other return codes than 200 OK might have other functions. A 303 response (http://en.wikipedia.org/wiki/HTTP_303) could have a function like 'mg_send_response_forward(conn, location)', ...

... the idea is not completely elaborated now.

**Comment by cdbishop** [comment]
That sounds nicer to use than manually specifying the code & message and it removes the possibility of something like a 200 Bad Request response which could be done with my suggestion.

**Pull request closed without merging**